xonsh/jobs.py

```diff
@@ -61,7 +61,14 @@ def _kill(job):
         _send_signal(job, signal.SIGKILL)
 
     def _send_signal(job, signal):
-        os.killpg(job['pgrp'], signal)
+        if ON_DARWIN:
+            # This is kind of workaround for OSX, since os.killpg() may
+            # cause PermissionError, we still need to figure out why.
+            # see Github issue #1012
+            for pid in job['pids']:
+                os.kill(pid, signal)
+        else:
+            os.killpg(job['pgrp'], signal)
```

Comment:
Can you maybe change this around a bit so that there are two `_send_signal()` functions defined.  One if we are one, darwin and another one, otherwise.  Maybe:

```python

if ON_DARWIN:
    def _send_signal(...):
        # This is kind of workaround for OSX, since os.killpg() may
        # cause PermissionError, we still need to figure out why.
        # see Github issue #1012
        for pid in job['pids']:
           os.kill(pid, signal)
else:
    def _send_signal(...)
        os.killpg(job['pgrp'], signal)
```

I think this provides a little better isolation of the problem.